File: solutions/opus-1/opus/python/roman.py
import sys

VALS = [(1000, "M"), (900, "CM"), (500, "D"), (400, "CD"), (100, "C"), (90, "XC"),
        (50, "L"), (40, "XL"), (10, "X"), (9, "IX"), (5, "V"), (4, "IV"), (1, "I")]

out = []
for tok in sys.stdin.read().split():
    n = int(tok)
    s = []
    for v, r in VALS:
        while n >= v:
            s.append(r)
            n -= v
    out.append("".join(s))
sys.stdout.write("\n".join(out) + ("\n" if out else ""))

File: solutions/opus-1/opus/javascript/roman.js
const data = require('fs').readFileSync(0, 'utf8');
const vals = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
const syms = ['M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I'];
const out = [];
for (const tok of data.split(/\s+/)) {
  if (!tok) continue;
  let n = parseInt(tok, 10);
  let s = '';
  for (let i = 0; i < vals.length; i++) {
    while (n >= vals[i]) { s += syms[i]; n -= vals[i]; }
  }
  out.push(s);
}
process.stdout.write(out.length ? out.join('\n') + '\n' : '');
